spec/support/test_db.rb

```diff
@@ -0,0 +1,132 @@
+class TestDb
+
+  DATABASES = %w{hr arunit}
+
+  def self.build
+    db = self.new
+    db.drop_databases(DATABASES)
+    db.create_databases(DATABASES)
+    db.connection.logoff
+  end
+
+  def self.database_version
+     db = self.new
+     db.database_version
+  end
+
+  def self.drop
+    db = self.new
+    db.drop_databases(DATABASES)
+    db.connection.logoff
+  end
+
+  def connection
+    unless defined?(@connection)
+      begin
+        Timeout::timeout(5) {
+          if defined?(JRUBY_VERSION)
+            @connection = java.sql.DriverManager.get_connection(
+              'jdbc:oracle:thin:@127.0.0.1:1521/XE',
+              'system',
+              'oracle'
+            );
+          else
+            @connection = OCI8.new(
+              'system',
+              'oracle',
+              '(DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=127.0.0.1)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=XE)))'
+            )
+          end
+        }
+      rescue Timeout::Error
+        raise "Cannot establish connection with Oracle database as SYSTEM user. Seams you need to start local Oracle database"
+      end
+    end
+    @connection
+  end
+
+  def drop_databases(databases=[])
```

Comment:
It's not actually dropping a database.
Can you consider a different method name? Maybe 
setup_databases